src/compute/src/lib.rs

```diff
@@ -44,14 +44,14 @@ pub enum AsyncStackTraceOption {
 #[derive(Parser, Clone, Debug)]
 pub struct ComputeNodeOpts {
     // TODO: rename to listen_address and separate out the port.
+    /// The address for this service to listen to locally
     #[clap(long, default_value = "127.0.0.1:5688")]
-    pub host: String,
+    pub listen_address: String,
 
-    /// The address of the compute node's meta client.
-    ///
+    /// The address for contacting this instance of the compute service.
```

Comment:
What about adding more detailed explanations about which address is for what services and contacted by whom?

Reply:
Valid point. StreamManager for instance uses it to compare against `ActorInfo.host` (I believe this is derived from `RegisterWorkerRequest`, which currently identifies itself via its `contact_address`).

Realistically, it's more of `public_address`, or `identifying_address`. 

While `host` could be a valid name for this, I think its confusing.

> Hmmm, is listen_address also an address for "contacting this instance"

Not actually. This listens to a port locally. Whereas `contact_address` can be behind a docker container or proxy.

Reply:
I modified the param description to be a little clearer
```
    /// The address that this service listens to.
    /// Usually the localhost + desired port.
    #[clap(long, default_value = "127.0.0.1:5688")]
    pub listen_address: String,

    /// The address for contacting this instance of the service.
    /// This would be synonymous with the service's "public address"
    /// or "identifying address".
    /// Optional, we will use listen_address if not specified.
    #[clap(long)]
    pub contact_address: Option<String>,
```

I hope it would be sufficiently clear.